tuplex/core/include/DataSet.h

```diff
@@ -263,11 +271,11 @@ namespace tuplex {
         // these are actions that cause execution
         virtual std::shared_ptr<ResultSet> collect(std::ostream &os = std::cout);
 
-        virtual std::shared_ptr<ResultSet> take(int64_t numElements, std::ostream &os = std::cout);
+        virtual std::shared_ptr<ResultSet> take(size_t topLimit, size_t bottomLimit, std::ostream &os = std::cout);
```

Comment:
add doxygen string to describe what this function does + explain edge case of overlap and what the program will do then. In addition, explain special values for everything.